# HG changeset patch
# User Gene Smith <gds@chartertn.net>
# Date 1709832030 -3600
# Node ID 0d90f48314f583c9d23ae53dfca62e4e68951454
# Parent  73aeb8ccaea80e456e384952a5c09e8e75359ddb
Bug 1863820 - Restore mailnews.mark_message_read.<servertype> functionality removed in 115. r=mkmelin

Now you can again create a new pref, e.g., mailnews.mark_message_read.rss set true so that
when any rss folder is exited, all messages are marked read. This capability was removed
from 115 for unknown reasons. In addition to rss folders, you can make this apply to
other folder types by setting the pref ending to nntp, imap, none (for Local Folders),
and pop3. Multiple prefs can be created to apply this to more than one folder type.
TODO: Consider retiring this feature in favor of an folders.onLeavingFolder API event
(or something similar) that add-ons could hook into.

Differential Revision: https://phabricator.services.mozilla.com/D196643

Note for ESR 115: The TB folks didn't backport this correctly and left this hunk in :-(

diff --git a/mail/base/content/folderDisplay.js b/mail/base/content/folderDisplay.js
--- a/mail/base/content/folderDisplay.js
+++ b/mail/base/content/folderDisplay.js
@@ -879,20 +879,8 @@ FolderDisplayWidget.prototype = {
     return passwordPromptRequired;
   },
 
   /**
-   * Let the viewWrapper know if it should mark the messages read when leaving
-   *  the provided folder.
-   *
-   * @returns true if the preference is set for the folder's server type.
-   */
-  shouldMarkMessagesReadOnLeavingFolder(aMsgFolder) {
-    return Services.prefs.getBoolPref(
-      "mailnews.mark_message_read." + aMsgFolder.server.type
-    );
-  },
-
-  /**
    * The view wrapper tells us when it starts loading a folder, and we set the
    *  cursor busy.  Setting the cursor busy on a per-tab basis is us being
    *  nice to the future. Loading a folder is a blocking operation that is going
    *  to make us unresponsive and accordingly make it very hard for the user to
